public/app/features/query/components/QueryActionComponent.ts

```diff
@@ -0,0 +1,18 @@
+import { DataQuery } from '@grafana/data';
+
+export interface QueryActionComponentProps {
+  query?: DataQuery;
+  queries?: DataQuery[];
+}
+
+export type QueryActionComponent = React.ComponentType<QueryActionComponentProps>;
+
+let extraRenderQueryActions: QueryActionComponent[] = [];
+
+export function addExtraRenderQueryAction(extra: QueryActionComponent) {
+  extraRenderQueryActions = extraRenderQueryActions.concat(extra);
+}
+
+export function getAllExtraRenderQueryAction(): QueryActionComponent[] {
+  return extraRenderQueryActions;
+}
```

Comment:
I think we should mark this as internal
https://github.com/grafana/grafana/blob/0383becc46d9e1171919b05ecae0e798a7d202c0/public/app/features/dashboard/state/DashboardModel.ts#L254